sdk/search/azure-search/src/test/java/com/azure/search/SearchTestBase.java

```diff
@@ -121,17 +124,21 @@ void assertRangeFacets(List<RangeFacetResult> baseRateFacets, List<RangeFacetRes
         Assert.assertEquals(2, lastRenovationDateFacets.get(1).getCount().intValue());
     }
 
-    List<RangeFacetResult> getRangeFacetsForField(
+    @SuppressWarnings("unchecked")
+    <T> List<RangeFacetResult<T>> getRangeFacetsForField(
         Map<String, List<FacetResult>> facets, String expectedField, int expectedCount) {
         List<FacetResult> facetCollection = getFacetsForField(facets, expectedField, expectedCount);
-        return facetCollection.stream().map(RangeFacetResult::new)
+        return facetCollection.stream().map(facetResult -> new RangeFacetResult<T>(facetResult.getCount(),
```

Comment:
The logic to create `RangeFacetResult` and `ValueFacetResult` from a `FacetResult` should be part of the SDK, not just the tests. Every customer that uses facets will need to do this, so we might as well make it easy.

Reply:
Do we want to add a public API in FacetResult to get RangeFacetResult/ValueFacetResult?

Reply:
FacetResult is auto-gen code. For the effort of getting RangeFacetResult/ValueFacetResult from FacetResult, we have to change the swagger definition.

Reply:
By "change the swagger definition" I assume you mean use an AutoRest transform directive? Because there's no way to express local language-specific methods on a Swagger type. I'm assuming you will handle that, and our team will own the actual Swagger definition of the REST API.

Reply:
I might misunderstood your statement before.

I currently added back the constructor of taking FacetResult inside of RangeFacetResult/ValueFacetResult. 

Let me know if furthermore functionalities needed here.

Reply:
It's definitely a step in the right direction. I'm not sure what more we can do to make this more discoverable. @tg-msft had similar concerns about this pattern for the new .NET SDK. Maybe we can find a common solution for Preview 2.

Reply:
Tracking with https://github.com/Azure/azure-sdk-for-net/issues/10613